geoh5py/groups/interpretation_section.py

```diff
@@ -0,0 +1,411 @@
+# #  GNU Lesser General Public License for more details.
+# #
+# #  You should have received a copy of the GNU Lesser General Public License
+# #  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+#
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+from uuid import UUID
+
+from pydantic import BaseModel, model_validator
+
+from ..shared.utils import str2uuid
+from .base import Group
+
+
+if TYPE_CHECKING:  # pragma: no cover
+    from ..objects import Curve, Slicer
+
+ATTRIBUTE_MAP = {
+    "Normal X": "normal_x",
+    "Normal Y": "normal_y",
+    "Normal Z": "normal_z",
+    "Position X": "position_x",
+    "Position Y": "position_y",
+    "Position Z": "position_z",
+}
+
+
+class InterpretationSectionParams(BaseModel):
+    """
+    The class to store the interpretation section.
+
+    :param normal_x: The normal x component.
+    :param normal_y: The normal y component.
+    :param normal_z: The normal z component.
+    :param position_x: The position x component.
+    :param position_y: The position y component.
+    :param position_z: The position z component.
+    """
+
+    normal_x: float
+    normal_y: float
+    normal_z: float
+    position_x: float
+    position_y: float
+    position_z: float
+
+    def in_list(self, params_list: list[InterpretationSectionParams]) -> int | None:
+        """
+        Check if the parameters are in the list.
+
+        :param params_list: The list of parameters.
+
+        :return: The index of the parameters in the list else None.
+        """
+        for idx, param in enumerate(params_list):
+            if param.map_back == self.map_back:
+                return idx
+        return None
+
+    @model_validator(mode="before")
+    @classmethod
+    def map_attributes(cls, data: dict) -> dict:
+        """
+        Map the attributes to the right keys.
+
+        :param data: The data to map.
+
+        :return: The mapped data.
+        """
+        return {ATTRIBUTE_MAP.get(key, key): value for key, value in data.items()}
+
+    @property
+    def map_back(self) -> dict:
+        """
+        Map the attributes back to the original keys of attribute map.
+
+        :return: The mapped data.
+        """
+        return {key: getattr(self, value) for key, value in ATTRIBUTE_MAP.items()}
+
+    @staticmethod
+    def verify(
+        params: dict | InterpretationSectionParams,
+    ) -> InterpretationSectionParams:
+        """
+        Verify if the parameters are correct.
+
+        :param params: The parameters to verify.
+
+        :return: An instance of the class.
+        """
+        if isinstance(params, dict):
+            return InterpretationSectionParams(**params)
+        if not isinstance(params, InterpretationSectionParams):
+            raise TypeError(
+                "Interpretation section must be an InterpretationSectionParams"
+                f"or a dict, {type(params)} provided."
+            )
```

Comment:
Also seems out of place. This kind of validation should belong to the class holding the sections